### DISCUSSION TITLE
Retrieve the system's time zone

### DISCUSSION THREAD

**Discussion by GabrielSieben** [NativePHP/laravel#181] [Ideas]
Hello,

I would appreciate if there was a feature where I could easily determine the user's system time zone, so that I can adjust dates and times in my app accordingly.

Gabriel

**Comment by osc2nuke** [comment]
Easiest via vanilla javascript: 
 ```    
<script>
            console.log( Intl.DateTimeFormat().resolvedOptions().timeZone );
</script>
```
Console output: Europe/Amsterdam

Many prob will laugh, but in the early days we just did:
```
            <script type="text/javascript">
                  var TimeZone=  Intl.DateTimeFormat().resolvedOptions().timeZone ;
                   <?php $timeZone= "<script>document.write(TimeZone)</script>"?>   
            </script>
            <?php echo $timeZone;?>
  ```

**Reply by simonhamp** [reply]
Yeh, slightly tricky to use this in PHP without some extra hops.

We'd need a way to instruct PHP about the system's current timezone when it's booted up as it doesn't seem to receive this from the environment by default.

You can get a form of timezone on a mac with the command: `date +'%Z'` (I guess it's the same on Linux? Not sure about Windows)

So totally doable

**Comment by basst85** [comment]
@simonhamp Maybe this can be added to **electron-plugin**? Using ``webContents.executeJavaScript('Intl.DateTimeFormat().resolvedOptions().timeZone').then((result) => {
    ...
  })``
  
  (or return the whole response object from method ``resolvedOptions()``, because it also contains other interesting locale system info)

**Reply by simonhamp** [reply]
This is not my preferred solution.

If we can pass the timezone via an environment variable into the PHP process as it runs, then we're not having to make calls over to an Electron window to fetch it.

**Comment by simonhamp** [comment]
This was [added](https://github.com/NativePHP/laravel/blob/main/src/System.php#L78-L89) recently. [Docs here](https://nativephp.com/docs/1/the-basics/system#time-zones).
